src/views/darkroom.c

```diff
@@ -211,7 +211,21 @@ void expose(
     else
       dt_gui_gtk_set_source_rgb(cr, DT_GUI_COLOR_DARKROOM_BG);
     cairo_paint(cr);
-    cairo_translate(cr, .5f * (width - wd), .5f * (height - ht));
+
+    float closeup2 = 1.0;
+    const float scale = dt_dev_get_zoom_scale(dev, zoom, 1.0f, 0) * darktable.gui->ppd;
+
+    if (scale > 1)
+    {
+      closeup2 = scale;
+    }
+
+    cairo_translate(cr, .5f * (width - wd*closeup2), .5f * (height - ht*closeup2));
+    if(closeup2 > 1)
+    {
+      cairo_scale(cr, closeup2, closeup2);
```

Comment:
So it will produce less pixels, and leave upscaling up to cairo, if i'm reading this right?

Reply:
Yes. At least this was what i was trying to accomplish.

Reply:
.. and since we pass 8-bit integers to cairo, the scaling will be kinda horrible.
I fail to see how is this better.

Reply:
Perhaps I don't understand it right. But is there any benefit to calculate all modules on an upsamples image, instead of applying it to the full (100%) resolution und then just scaling the output? 
I think there is no valuable information added by upscaling.

Reply:
Even so, cairo scaling will be worse quality-wise.

Reply:
If there is better option for the final scaling I have no problem with it.
But applying filters on pixels that were not in the original images still makes no sense to me.

imho at 200% zoom it should just display the pixels at twice the size and not necessarily smooth them out. The preview quality should not be better then the exported 100% image.

Reply:
I would even argue that worse interpolation done by cairo, namely nearest neighbour, would be the best approach. At least if we can guarantee that we are displaying the image at 200% and not some float factor.

Reply:
What would be a better approach?
Split the current scaling to before and after the other modules?